src/settings.cpp

```diff
@@ -1159,6 +1190,24 @@ static void SaveVersionInConfig(IniFile &ini)
 	group->GetOrCreateItem("ini_version").SetValue(std::to_string(INIFILE_VERSION));
 }
 
+/**
+ * Save BaseGraphics set selection and configuration.
+ */
+static void GraphicsSetSaveConfig(IniFile &ini)
+{
+	const GraphicsSet *used_set = BaseGraphics::GetUsedSet();
+	if (used_set == nullptr) return;
+	ini.RemoveGroup("graphicsset");
+	IniGroup *group = ini.GetGroup("graphicsset");
+	group->GetOrCreateItem("name").SetValue(used_set->name);
+	group->GetOrCreateItem("shortname").SetValue(fmt::format("{:08X}", BSWAP32(used_set->shortname)));
+	const GRFConfig *extra_cfg = used_set->GetExtraConfig();
+	if (extra_cfg != nullptr && extra_cfg->num_params > 0) {
+		group->GetOrCreateItem("extra_version").SetValue(fmt::format("{}", extra_cfg->version));
+		group->GetOrCreateItem("extra_params").SetValue(GRFBuildParamList(extra_cfg));
+	}
```

Comment:
Could perhaps do with some extra newlines to break it into more readable sections...